nemo_curator/stages/synthetic/qa_multilingual_synthetic.py

```diff
@@ -0,0 +1,149 @@
+        responses = []
+        for i in range(self.num_samples):
+            print(f"Generating sample {i+1}/{self.num_samples} (sync)...")
+            language = secrets.choice(self.languages)
+            prompt = self.prompt.format(language=language)
+            response = self.client.query_model(
+                model=self.model_name,
+                messages=[{"role": "user", "content": prompt}],
+                generation_config=self.generation_config
+            )
+            generated_text = self._process_llm_response(response)
+            responses.append(generated_text)
+        return responses
+
+    def _process_async(self) -> list[str]:
+        """Process samples using async client (concurrent).
+
+        This method handles both cases:
+        - Normal case: No event loop exists, creates one with asyncio.run()
+        - Edge case: Called from async context, runs in separate thread
+        """
+        try:
+            # Check if we're already in an async context
+            asyncio.get_running_loop()
+        except RuntimeError:
+            # No loop running - this is the expected/normal case
+            # Safe to use asyncio.run() which creates its own loop
+            return asyncio.run(self._generate_responses_async())
+
+        # If we get here, there's already a loop running
+        # This is an edge case (e.g., Ray async actors), but we can handle it
+        # by running in a new thread with its own loop
+        import concurrent.futures
+        with concurrent.futures.ThreadPoolExecutor(max_workers=1) as executor:
+            future = executor.submit(asyncio.run, self._generate_responses_async())
+            return future.result()
+
+    async def _generate_responses_async(self) -> list[str]:
+        """Generate responses asynchronously using concurrent requests."""
+        async def generate_single_response(_i: int) -> str:
+            language = secrets.choice(self.languages)
+            prompt = self.prompt.format(language=language)
+            response = await self.client.query_model(
+                model=self.model_name,
+                messages=[{"role": "user", "content": prompt}],
+                generation_config=self.generation_config
+            )
+            return self._process_llm_response(response)
+
+        # Create tasks for all samples and execute concurrently
+        tasks = [generate_single_response(i) for i in range(self.num_samples)]
+        return await asyncio.gather(*tasks)
+
+
+class LanguageFilter(DocumentFilter):
+    """Filter documents based on language prefix codes.
+
+    Keeps documents that start with any of the specified language codes.
+    Designed to work with prompts that instruct LLMs to prefix responses
+    with language codes (e.g., [EN], [FR], [DE]).
+    """
+
+    def __init__(self, languages: list[str]):
+        self._name = "language_filter"
+        self.languages = languages
+
+
+    def score_document(self, text: str) -> float:
```

Comment:
Surprised the linter didn't catch this, I think there is an extra line here that should be removed.